bigquery/start.py

```diff
@@ -0,0 +1,224 @@
+import os.path
+import requests, json
+
+from google.auth.transport.requests import Request
+from google.oauth2.credentials import Credentials
+from google_auth_oauthlib.flow import InstalledAppFlow
+from googleapiclient.discovery import build
+from googleapiclient.errors import HttpError
+
+# If modifying these scopes, delete the file token.json.
+SCOPES = ["https://www.googleapis.com/auth/bigquery"]
+
+WEATHER_API_KEY = "YOUR_KEY"
+
+
+def create_dataset(creds, project_id, dataset_id):
+    try:
+        service = build("bigquery", "v2", credentials=creds)
+        service.datasets().insert(
+            projectId=project_id,
+            body={
+                "datasetReference": {
+                    "projectId": project_id,
+                    "datasetId": dataset_id,
+                }
+            },
+        ).execute()
+    except HttpError as err:
+        print(err)
```

Comment:
Print statements have their place, but I generally avoid them in production code. Prefer logging instead, which has a broader scope -and can be used to print to your CLI if that is the outcome you're looking for.
The other comment at this point is that you are using http requests, but Google Cloud offers client libraries for its services, e.g. there is a BigQuery Python client library that will offer the equivalent of your create_dataset function out of the box.

Reply:
I think the difference would be where we start our BigQuery. If we run the Python code locally, we need this code. If we run the Python code in the cloud shell, we will use the BigQuery Python client library. If we are using the 2nd option, I will update my code.